Allow entering some of the mode and frequency commands

linustolke opened the pull request:
N1MM has the option to enter the frequency or the mode in the callsign field and press return. That will make the rig jump to that mode or frequency. A similar, but most likely simpler - I did not experiement on the nitty gritty details of N1MM, is added by this change.


PS. I just found not1mm today and I must say that this is a very interesting project for me. I think am on just the right level of contesting, python, open source and Linux to appreciate this. Just by looking at the code, I am learning a lot about how to build and application with Qt that I previously knew nothing about. When participating in contests at the club, we use N1MM but should I get antennas up at home, for some QRP contesting, I will probably now use Not1mm, I have failed to install N1MM in wine.

mbridak commented:
Not1mm allows frequency and mode changes via the callsign field. You just press the spacebar instead of the enter key. But I'll see how the change works with it.

linustolke commented:
Thanks for the nice answer.

I am sorry, I missed that function. I have to admit I didn't spend that much time searching for the description on how the input field is processed. N1MM performs these changes (frequency and mode) when Enter is pressed.

I now found the description in the N1MM doc, and I realize that I didn't implement the decimal separator so this implementation is restricted to full kHz. I also don't do the relative +/- frequency changes. This is described in [Callsign Box Text Commands](https://n1mmwp.hamdocs.com/manual-windows/entry-window/#callsign-box-text-commands).

linustolke commented:
On the other hand, I don't think pressing Enter in another field in N1MM will perform the command but my implementation does.

To test my implementation: enter "123" in the calls field without pressing Space or Enter. Klick in the CQ Zone (I testing with the CQWW test) to enter that field, then press Enter. Then the command "123" is issued jumping to the frequency 14123.

I am not sure if this is good or bad. It is a strange case. I added a hotkeys explanation in a way that it is correctly describing this (and fixed a bug in the initial commit).

mbridak commented:
closing as already implimented.
